barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -126,15 +170,14 @@ void AvmTraceBuilder::op_add(
 {
     auto clk = static_cast<uint32_t>(main_trace.size()) + 1;
 
-    auto const res = resolve_ind_three(call_ptr, clk, indirect, a_offset, b_offset, dst_offset);
-    bool tag_match = res.tag_match;
+    // Resolve any potential indirects in the order they are encoded in the indirect byte.
+    auto [resolved_a, resolved_b, resolved_c] = resolve_indirects<3>(indirect, { a_offset, b_offset, dst_offset });
```

Comment:
totally optional: C++ is quite good at inferring template args. Removing the `<3>` or using `<>` might works. (but not sure, try it)

Reply:
unfortunately it doesnt work, i think it cant infer based on the `{ a_offset, b_offset, dst_offset }` being passed directly to the call.

I can get the inference to work if i do the following first
`std::array<uint32_t>3 offsets = { a_offset, b_offset, dst_offset };`

Reply:
Don't worry, leave it as is!